app/models/contact.rb

```diff
@@ -1,4 +1,18 @@
 class Contact < ApplicationRecord
   validates :salesforce_id, presence: true, uniqueness: true
+  has_many :subscriptions
+  has_many :lists, through: :subscriptions
 
+  def list_subscriptions
+    @list_subscriptions = []
+    # Add all the lists to the subscriptions array, we update subscribed to true below if the user is subscribed.
+    # This reduces calls to the API by providing all available lists in the users API.
+    List.all.each do |list|
+      @list_subscriptions.push({ id: list.pardot_id,
+                                 title: list.title,
+                                 description: list.description,
+                                 subscribed: subscriptions.any? { |subscription| subscription.list == list } })
+    end
+    @list_subscriptions
```

Comment:
1. This code has functionality not related to `Contact` (the listing of all Lists), which seems wrong.  Also the array being built here is the contents you are trying to get out in an API, which is also funky from a model-shouldn't-care-about-the-API outputs perspective.
2. I see the "this reduces calls to the API" comment, which is yeah true, but I think one extra call to the Lists index API (which exists) is better than this network request optimization code.  JS client calls /api/v1/lists and /api/v1/user and does the logic to show a bunch of lists and the ones the user is subscribed to.
3. Then this whole method goes away and your users controller line that has `subscriptions: list_subscriptions` changes to `subscriptions: current_contact.subscriptions`.

If you still want this code, put it in a controller.